crates/modules/rules-engine/src/engine.rs

```diff
@@ -90,7 +90,7 @@ impl PulsarEngine {
                 for r in ruleset.matches(event) {
                     self.internal
                         .sender
-                        .send_threat_derived(event, r.rule.name.clone(), None);
+                        .send_threat_derived(event, r.name.clone(), None);
```

Comment:
Rule `Metadata` must not be removed, instead you should use it here:

```rust
                for r in ruleset.matches(event) {
                    let metadata_value =
                        Value::try_from(&r.metadata).expect("impossibile to serialize metadata");

                    self.internal.sender.send_threat_derived(
                        event,
                        r.rule.name.clone(),
                        Some(metadata_value),
                    );
                }
```

You have to derive `Serialize` for `Metadata` to use it in `Value:try_from`. 

PS: `Value::try_from` needs a small change, now take a `T`, but it should take a reference.

Reply:
OK, I thought that metadata doesn't matter for compiled rules (since it's not patch of the match), but that makes sense.